core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.kt

```diff
@@ -104,6 +105,11 @@ class SearchActivity : BaseActivity() {
     overridePendingTransition(anim.fade_in, anim.fade_out)
   }
 
+  override fun onPrepareOptionsMenu(menu: Menu?): Boolean {
+    menu?.findItem(R.id.menu_searchintext)?.isVisible = !isFromTabSwitcher()
```

Comment:
there is already a field for this menuItem but also this is breaking the architecture of this screen. You should emit an action that mutates the state so that this menuItem is hidden by the state of this screen

Reply:
Could you clarify what's meant by emitting an action ?

Reply:
the viewmodel has a field `actions`, you offer a member of the Action class to this, it takes this and then depending on what it wants to do with it augments the state or emits a sideeffect.

[This article](https://www.raywenderlich.com/817602-mvi-architecture-for-android-tutorial-getting-started) might give you a starting point.

The idea is the visibility of this thing needs to be in the state. The item will be hidden in the `render` method if it all works out